src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1126,6 +1126,11 @@ private bool HasInstanceFields()
                     switch (m.Kind)
                     {
                         case SymbolKind.Field:
+                            if (m is TupleErrorFieldSymbol)
+                            {
+                                break;
+                            }
```

Comment:
Is this needed?

Reply:
For empty structs, we emit some layout information (`[StructLayout(LayoutKind.Sequential, Size = 1)]`).
Without this change, we don't emit this layout information correctly, and so PEVerification fails (struct must have a type or a size).

---
In reply to: [425349126](https://github.com/dotnet/roslyn/pull/44231#discussion_r425349126) [](ancestors = 425349126)

Reply:
Why do we have an error field in a valid struct?

---
In reply to: [425410620](https://github.com/dotnet/roslyn/pull/44231#discussion_r425410620) [](ancestors = 425410620,425349126)

Reply:
Because a type `System.ValueTuple'2` is recognized as a tuple, and therefore has two fields `Item1` and `Item2` that can be accessed regardless of what the source says. We make them error fields because they lack real/backing fields.

Similarly, on an 8-tuple, we have an `Item8` (an error field) if `ValueTuple'1` lacks an `Item1`.

---
In reply to: [425411648](https://github.com/dotnet/roslyn/pull/44231#discussion_r425411648) [](ancestors = 425411648,425410620,425349126)